File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        Dictionary<int, int> count = new Dictionary<int,int>();
        int test = nums.Length / 2;
        foreach(var num in nums){
            if(!count.ContainsKey(num)) count[num] = 1;
            else count[num]++;
            if(count[num] > test) return num;
        }
        return -1;
        
    }
}

File: 0169-majority-element/0169-majority-element.ts
function majorityElement(nums: number[]): number {
    const map = new Map<number, number>();
    const n = nums.length;
    for(let num of nums){
        const count = (map.get(num) || 0) +1;
        map.set(num, count);
        if(count > Math.floor(n/2)) return num;
    }
    return -1;

};